File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int current_max = 0;
        int omax = Integer.MIN_VALUE;

        for (int i=0;i<nums.length;i++){
            if (current_max < 0){
                current_max = nums[i];
            }
            else{
                current_max = current_max + nums[i];
            }

            omax = Math.max(omax,current_max);
        }

        return omax;
    }
}

File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int n = nums.size();
     

        int maxSum = INT_MIN;
        int csum = 0;

        for(int i = 0; i < n; i++){
            csum += nums[i];
            maxSum = max(maxSum, csum);

            if (csum < 0) csum = 0;    
        }

        return maxSum;
    }
};